### PR TITLE
PAN-15 Save zones during Site storing

### PR DESCRIPTION
**Pull request by teghnet**

### CODE REVIEW

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ public function render($request, Exception $exception)
                 return new JsonResponse([
                     $prev->getErrorCode(),
                     $prev->getMessage(),
-                    $prev->getTraceAsString()
+                    $prev->getTrace()
```
Comment (teghnet) [inline app/Exceptions/Handler.php:37]: For better debugging

File: app/Http/Controllers/Rest/SitesController.php [hunk 2]
```diff
@@ -32,14 +33,19 @@ public function add(Request $request)
     {
         $this->validateRequestObject($request, 'site', Site::$rules);
 
+        /** @var Site|Builder $site */
         $site = Site::create($request->input('site'));
         $site->user_id = Auth::user()->id;
-        $site->site_requires = $request->input('site.filtering.requires') ?? $request->input('site.targeting.requires');
-        $site->site_excludes = $request->input('site.filtering.excludes') ?? $request->input('site.targeting.excludes');
-
+        $site->site_requires = $request->input('site.filtering.requires');
```
Comment (teghnet) [inline app/Http/Controllers/Rest/SitesController.php:39]: @Meskat this removes `targeting` support in `sites` - only `filtering` remains

File: app/Http/Controllers/Rest/SitesController.php [hunk 3]
```diff
@@ -86,124 +92,21 @@ public function read(Site $site)
     }
 
     /**
-     * @param Request $request
-     *
-     * @return \Illuminate\Http\JsonResponse
-     *
-     * @throws \Adshares\Adserver\Exceptions\JsonResponseException
-     * @throws \Illuminate\Validation\ValidationException
+     * @deprecated
```
Comment (teghnet) [inline app/Http/Controllers/Rest/SitesController.php:95]: this needs to be handled by a dedicated service

File: database/migrations/2018_04_06_103930_create_zones_table.php [hunk 4]
```diff
@@ -41,12 +41,12 @@ public function up()
             $table->string('name', 32);
             $table->integer('width');
             $table->integer('height');
+            $table->integer('status', Zone::STATUS_DRAFT);
 
-            $table->foreign('site_id')->references('id')->on('sites')->onUpdate('RESTRICT')->onDelete('CASCADE');
+            $table->foreign('site_id')->references('id')->on('sites')->onUpdate('RESTRICT')->onDelete('RESTRICT');
```
Comment (teghnet) [inline database/migrations/2018_04_06_103930_create_zones_table.php:46]: @PawelPodkalicki Do you agree?

File: routes/rpc.php [hunk 5]
```diff
@@ -35,9 +34,10 @@
 // tmp mocked solutions
 Route::post('chart', [ChartsController::class, 'chart']);
 Route::get('options/campaigns/targeting', [CampaignsController::class, 'targeting']);
-Route::get('options/sites/targeting', [SitesController::class, 'targeting']);
+Route::get('options/sites/targeting', [Simulator::class, 'targeting']);
 Route::post('publisher_chart', [ChartsController::class, 'publisherChart']);
-Route::get('config/banners', [SitesController::class, 'banners']);
+Route::get('config/banners', [Simulator::class, 'zoneTypes']);//@deprecated
```
Comment (teghnet) [inline routes/rpc.php:40]: @Meskat Will be removed after PAN-49 (not urgent)

File: app/Exceptions/Handler.php (hunk 1 above)
Comment (c3zi) [reply app/Exceptions/Handler.php:37]: If we'are doing anything in this place we should check whether an env is PROD or DEV at least ... Could you add this checking point to disable displaying all debug information in prod?

File: app/Http/Controllers/Rest/SitesController.php [hunk 6]
```diff
@@ -32,14 +33,19 @@ public function add(Request $request)
...
-        $site->site_excludes = $request->input('site.filtering.excludes') ?? $request->input('site.targeting.excludes');
-
+        $site->site_requires = $request->input('site.filtering.requires');
+        $site->site_excludes = $request->input('site.filtering.excludes');
         $site->save();
 
-        $response = self::json(compact('site'), 201);
+        $adUnits = $request->input('site.ad_units');
+        $adUnits = $this->mapAdUnitsToZoneModel($adUnits);//needs to be removed after front refactor
+
```
Comment (c3zi) [inline app/Http/Controllers/Rest/SitesController.php:45]: `$zones = $this->mapAdUnitsToZoneModel($adUnits);` ??

File: app/Models/Zone.php [hunk 7]
```diff
@@ -25,15 +25,42 @@
...
+        '120x240',
+        '200x200',
+        '300x1050',
+        '250x250',
+        '320x50',
+        '970x90',
+        '970x250',
+        '750x100',
+        '750x200',
+        '750x300',
```
Comment (c3zi) [inline app/Models/Zone.php:52]: Until the same values in the same order won't be available in FRONT it does not work. So please set my old values at the beginning to compatible with the front.

**Review by c3zi — CHANGES_REQUESTED** [review]

File: routes/rpc.php [hunk 8]
```diff
@@ -35,9 +34,10 @@
 // tmp mocked solutions
 Route::post('chart', [ChartsController::class, 'chart']);
 Route::get('options/campaigns/targeting', [CampaignsController::class, 'targeting']);
-Route::get('options/sites/targeting', [SitesController::class, 'targeting']);
+Route::get('options/sites/targeting', [Simulator::class, 'targeting']);
```
Comment (c3zi) [inline routes/rpc.php]: `Route::get('options/sites/targeting', [Simulator::class, 'targeting']);`
-->
`Route::get('options/sites/filtering', [Simulator::class, 'targeting']);`

??

File: app/Models/Zone.php (hunk 7 above)
Comment (teghnet) [reply app/Models/Zone.php:52]: They are. I changed front as well.

File: app/Exceptions/Handler.php (hunk 1 above)
Comment (teghnet) [reply app/Exceptions/Handler.php:37]: The `APP_DEBUG=false` in `.env` protects against that.
When set to `false`, api returns:
```json
{
    "message": "Server Error"
}
```

**Review by c3zi — APPROVED** [review]

**Pull request merged**